File: app/src/main/java/com/liarstudio/courierservice/Activities/MainActivity.java
package com.liarstudio.courierservice.Activities;

import android.content.Intent;
import android.content.SharedPreferences;
import android.support.annotation.NonNull;
import android.support.design.widget.NavigationView;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentTransaction;
import android.support.v4.view.GravityCompat;
import android.support.v4.widget.DrawerLayout;
import android.support.v7.app.ActionBarDrawerToggle;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.preference.PreferenceManager;
import android.support.v7.widget.Toolbar;
import android.view.MenuItem;
import android.view.View;
import android.widget.TextView;

import com.liarstudio.courierservice.API.ApiUtils;
import com.liarstudio.courierservice.Adapters.PagerAdapterMy;
import com.liarstudio.courierservice.Adapters.PagerAdapterNew;
import com.liarstudio.courierservice.Fragments.HomeFragment;
import com.liarstudio.courierservice.Fragments.SettingsFragment;
import com.liarstudio.courierservice.Fragments.PackageFragment;
import com.liarstudio.courierservice.R;
import com.liarstudio.courierservice.BaseClasses.Package;

import org.w3c.dom.Text;

import static com.liarstudio.courierservice.API.ApiUtils.CURRENT_USER;

public class  MainActivity extends AppCompatActivity
    implements NavigationView.OnNavigationItemSelectedListener {

    /*
    ****** STATIC CONSTANT AREA ******
    */

    public static final int REQUEST_ADD_OR_EDIT = 1;
    public static final int REQUEST_MAP = 2;


    public static final String VOL_COEFFICIENT = "size_dimensions"; //"vol_coefficient";
    public static final String WEIGHT_COEFFICIENT = "weight_dimensions";//"weight_coefficient";
    public static final String PREFERENCES_FILENAME = "preferences_data";
    public static final String ON_FIRST_LAUNCH = "first_launch";



    /*
    ****** FIELDS AREA ******
    */

    public static int managerType = 0;


    DrawerLayout drawer;
    Fragment fragment;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        loadCoefficients();

        managerType = 0;
        fragment = new HomeFragment();

        FragmentTransaction fragmentTransaction = getSupportFragmentManager().beginTransaction();
        fragmentTransaction.replace(R.id.content_main, fragment).commit();

        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);


        drawer = (DrawerLayout) findViewById(R.id.drawer_layout);
        ActionBarDrawerToggle toggle = new ActionBarDrawerToggle(
                this, drawer, toolbar, R.string.navigation_drawer_open, R.string.navigation_drawer_close);
        drawer.addDrawerListener(toggle);
        toggle.syncState();

        NavigationView navigationView = (NavigationView) findViewById(R.id.nav_view);
        navigationView.setNavigationItemSelectedListener(this);

        View header = navigationView.getHeaderView(0);
        TextView textViewNavHeader = (TextView) header.findViewById(R.id.textViewNavHeader);
        textViewNavHeader.setText(CURRENT_USER.getName());



        MenuItem nav_my = navigationView.getMenu().findItem(R.id.nav_my);
        MenuItem nav_new = navigationView.getMenu().findItem(R.id.nav_new);

        if (ApiUtils.IS_ADMIN) {
            nav_my.setTitle(R.string.nav_my_admin);
            nav_new.setVisible(false);
        } else {
            nav_my.setTitle(R.string.nav_my);
            nav_new.setVisible(true);
        }
    }


    @Override
    public void onActivityResult(int requestCode, int resultCode, Intent data) {
        //super.onActivityResult(requestCode,resultCode, data);
        //fragment.onActivityResult(requestCode, resultCode, data);

        for (Fragment frag : getSupportFragmentManager().getFragments()) {
            frag.onActivityResult(requestCode, resultCode, data);
        }


    }

    @Override
    public boolean onNavigationItemSelected(@NonNull MenuItem item) {

        Fragment fragment = null;
        switch (item.getItemId()) {
            case R.id.nav_my:
                managerType = 0;
                fragment = new HomeFragment();
                break;
            case R.id.nav_new:
                managerType = 1;
                fragment = new HomeFragment();
                break;
            case R.id.nav_settings:
                fragment = new SettingsFragment();
                break;
            case R.id.nav_logout:
                CURRENT_USER = null;
                startActivity(new Intent(this, AuthActivity.class));
                finish();
                return true;
        }
        FragmentTransaction ft = getSupportFragmentManager().beginTransaction();
        ft.replace(R.id.content_main, fragment).commit();
        drawer.closeDrawer(GravityCompat.START);
        return true;
    }


    public void replaceFragment(int type) {
        fragment = null;
        managerType = type;
        fragment = new HomeFragment();

        FragmentTransaction ft = getSupportFragmentManager().beginTransaction();
        ft.replace(R.id.content_main, fragment).commit();
    }

    //Считываем коэффициенты из Preferences
    void loadCoefficients() {
        SharedPreferences sharedPref = PreferenceManager.getDefaultSharedPreferences(this);
        Package.WEIGHT_PROGRAM_STATE = Integer.parseInt(sharedPref.getString(WEIGHT_COEFFICIENT, "0"));
        Package.SIZE_PROGRAM_STATE = Integer.parseInt(sharedPref.getString(VOL_COEFFICIENT, "0"));
    }

}

File: app/src/main/java/com/liarstudio/courierservice/ui/screen/main/MainActivity.kt
package com.liarstudio.courierservice.ui.screen.main

import android.app.Fragment
import android.content.Intent
import android.os.Bundle
import android.support.v4.graphics.drawable.DrawableCompat
import android.support.v4.view.GravityCompat
import android.support.v7.app.ActionBarDrawerToggle
import android.support.v7.widget.Toolbar
import android.widget.ImageView
import android.widget.TextView
import com.liarstudio.courierservice.R
import com.liarstudio.courierservice.ui.base.EXTRA_FIRST
import com.liarstudio.courierservice.ui.base.isVisible
import com.liarstudio.courierservice.ui.base.screen.LoadState
import com.liarstudio.courierservice.ui.base.screen.view.BaseActivity
import kotlinx.android.synthetic.main.activity_main.*
import javax.inject.Inject

class MainActivity : BaseActivity<MainScreenModel>() {

    @Inject
    lateinit var presenter: MainActivityPresenter

    lateinit var fragment: Fragment

    override fun requestPresenter() = presenter

    override fun getLayout() = R.layout.activity_main

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        //Создаем новый MainFragment с менеджером PagerAdapterMyPackages
        fragment = MainFragment().apply { arguments = Bundle().apply { putSerializable(EXTRA_FIRST, MainTabType.MY) } }
        val fragmentTransaction = fragmentManager.beginTransaction()
        fragmentTransaction.replace(R.id.content_main, fragment).commit()

        initViews()

        presenter.loadCoeffs()
    }


    public override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        fragment.onActivityResult(requestCode, resultCode, data)

    }

    override fun renderData(screenModel: MainScreenModel) {
        screenModel.user?.let {
            setHeader(it.name, it.isAdmin)
            setNavigationButtons(it.isAdmin)
        }
        screenModel.currentFragment?.let { replaceFragment(it) }
    }

    override fun renderState(loadState: LoadState) { /*без реализации*/ }

    private fun replaceFragment(fragment: Fragment) {
        val ft = fragmentManager.beginTransaction()
        ft.replace(R.id.content_main, fragment).commit()
        drawer_layout.closeDrawer(GravityCompat.START)
    }

    private fun initViews() {
        initToolbar()
        initDrawerButton()
        initNavigationView()
    }

    private fun initToolbar() {
        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        setSupportActionBar(toolbar)
    }

    private fun initDrawerButton() {
        val toggle = ActionBarDrawerToggle(
                this, drawer_layout, toolbar, R.string.navigation_drawer_open, R.string.navigation_drawer_close)
        drawer_layout.addDrawerListener(toggle)
        toggle.syncState()
    }

    private fun initNavigationView() {
        nav_view.setNavigationItemSelectedListener { presenter.openScreen(it.itemId)}
    }

    private  fun setHeader(userName: String, isAdmin: Boolean) {
        val header = nav_view.getHeaderView(0)
        val usernameTv = header.findViewById<TextView>(R.id.username_tv)
        usernameTv.text = userName
        if (isAdmin) {
            val adminModeTv = header.findViewById<TextView>(R.id.admin_mode_tv)
            adminModeTv.isVisible = true
            val logoIv = header.findViewById<ImageView>(R.id.logo_iv)
            DrawableCompat.setTint(logoIv.background.mutate(), getColor(R.color.colorAccent))
        }

    }

    private fun setNavigationButtons(isAdmin: Boolean) {
        val navMy = nav_view.menu.findItem(R.id.nav_my)
        val navNew = nav_view.menu.findItem(R.id.nav_new)

        if (isAdmin) {
            navMy.setTitle(R.string.nav_my_admin)
            navNew.isVisible = false
        } else {
            navMy.setTitle(R.string.nav_my)
            navNew.isVisible = true
        }

    }
}
